### PR TITLE
Change Client-FQDN to be a struct

### PR DESCRIPTION
**Pull request by jejones3141**
It is split out into its own dictionary and has tests.

### CODE REVIEW

**Comment by jejones3141** [comment]
Struct and tests updated.

File: share/dictionary/dhcpv4/dictionary.rfc4072 [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- text -*-
+# Copyright (C) 2022 The FreeRADIUS Server project and contributors
+# This work is licensed under CC-BY version 4.0 https://creativecommons.org/licenses/by/4.0
+# Version $Id$
+##############################################################################
+#
+#  RFC4072 - The Dynamic Host Configuration Protocol (DHCP) Client
```
Comment (alandekok) [inline share/dictionary/dhcpv4/dictionary.rfc4072:7]: It's RFC 4702.  RFC 4072 is "Diameter extensions for EAP"

**Comment by jejones3141** [comment]
Corrections to files (both names and content) made and pushed to the branch.

**Comment by alandekok** [comment]
I merged the changes manually.  We'd prefer to do this as one commit instead of a bunch of smaller ones.  It keeps the git history a bit more readable.

**Pull request closed without merging**